homeassistant/components/trafikverket_weatherstation/sensor.py

```diff
@@ -151,38 +145,44 @@ async def async_setup_entry(
 ) -> None:
     """Set up the Trafikverket sensor entry."""
 
-    web_session = async_get_clientsession(hass)
-    weather_api = TrafikverketWeather(web_session, entry.data[CONF_API_KEY])
+    coordinator: TVDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
     entities = [
         TrafikverketWeatherStation(
-            weather_api, entry.entry_id, entry.data[CONF_STATION], description
+            coordinator, entry.entry_id, entry.data[CONF_STATION], description
         )
         for description in SENSOR_TYPES
     ]
 
     async_add_entities(entities, True)
```

Comment:
```suggestion
    async_add_entities(TrafikverketWeatherStation(
            weather_api, entry.entry_id, entry.data[CONF_STATION], description
            coordinator, entry.entry_id, entry.data[CONF_STATION], description
        )
        for description in SENSOR_TYPES
    )
```
Black will complain about this suggestion.

Since these are coordinator entities, we shouldn't need to call `updated_before_add=True`